### PR TITLE
Add support for custom registry and None endpoint

### LINKED ISSUE #2: Add custom registry support and None endpoint
Some projects have more than one actix_web `Server` (for example, to separate public and private APIs).  In this case user might not want to expose metrics endpoint on public server and would like to have one only on private server. I suggest making endpoint optional:
```
 let public_metrics = PrometheusMetrics::new("public", None);
 let private_metrics = PrometheusMetrics::new("private", Some("/metrics"));
```

Also for projects with multiple actix_web servers it might be useful to have one metrics endpoint for the whole project => it might be useful to share one `Registry` instance among multiple middlewares:
```
let r = Registry::new();
let public_metrics =
        PrometheusMetrics::new_with_registry(r.clone(), "public", None).unwrap();
let private_metrics =
        PrometheusMetrics::new_with_registry(r.clone(), "private", Some("/metrics")).unwrap();
```

This way middleware can be used in projects with one or more actix_web `Server`s out of the box

### PR DESCRIPTION
**Pull request by nWacky**
Resolves #2 
Might be useful for bigger projects

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -115,6 +115,75 @@ fn main() -> std::io::Result<()> {
...
+                                        None
+                                    )
+                                    .unwrap();
+
+# if false {
+    let public_handle = thread::spawn(move || {
+        HttpServer::new(move || {
+            App::new()
+                .wrap(private_prom.clone())
+                .service(web::resource("/test").to(test1))
```
Comment (nlopes) [inline src/lib.rs]: I think the way this is exposed (without a bit more docs) can be confusing. You have a public handle that has a "private prom" middleware, for which said private_prom is named "private_api" with `/metrics` but the function responds "Everyone can see it".

That being the intent, it's hard to follow.

File: src/lib.rs [hunk 2]
```diff
@@ -178,17 +258,19 @@ impl PrometheusMetrics {
             &["endpoint", "method", "status"],
         )
         .unwrap();
-        registry
-            .register(Box::new(http_requests_duration_seconds.clone()))
-            .unwrap();
+        registry.register(Box::new(http_requests_duration_seconds.clone()))?;
```
Comment (nlopes) [inline src/lib.rs:262]: This is an interesting one. If this fails, should we just bomb out, so that we present the same "guarantees" as the `new` method?

File: src/lib.rs [hunk 3]
```diff
@@ -200,7 +282,11 @@ impl PrometheusMetrics {
     }
 
     fn matches(&self, path: &str, method: &Method) -> bool {
-        self.endpoint == path && method == Method::GET
+        if self.endpoint.is_some() {
```
Comment (nlopes) [inline src/lib.rs:286]: I'm personally more inclined to `self.endpoint.is_some() && self.endpoint.as_ref().unwrap() == path && method == Method::GET` _but_ I don't know if compiler optimisations could swap the order of the function calls and make this not equivalent to what you wrote. Not too fussed to keep as is due to that.

**Review by nlopes — CHANGES_REQUESTED** [review]
Thank you so much for the work in this PR, it is much appreciated!

File: src/lib.rs (hunk 1 above)
Comment (nWacky) [reply src/lib.rs]: My bad, should've named these functions not `test1` and `test2`
Yes, server with `public_prom` should return "Everyone can see it"

File: src/lib.rs (hunk 2 above)
Comment (nWacky) [reply src/lib.rs:262]: `register()` [returns error if this counter is already registered](https://docs.rs/prometheus/0.7.0/prometheus/struct.Registry.html#method.register) => it should not panic on `unwrap` in `new`, where new `Registry` is created

It should return `Error` with custom `Registry` if I try to do this:
```rust
let metrics1 = PrometheusMetrics::new_with_registry(registry.clone(), 
                                                    "private_api", 
                                                    Some("/metrics")) .unwrap();
let metrics2 = PrometheusMetrics::new_with_registry(registry.clone(), 
                                                    "private_api", 
                                                    Some("/metrics")) 
                                                    .unwrap(); // << it should panic here because "private_api_http_requests_total" already exists
```

File: src/lib.rs (hunk 3 above)
Comment (nWacky) [reply src/lib.rs:286]: I'm honestly not sure about this either

I could've done it like this, but [path may be an empty string](https://docs.rs/actix-web/1.0.5/actix_web/http/uri/struct.Uri.html#method.path)
```
self.endpoint.unwrap_or("") == path && method == Method::GET
```
